src/php/settings/version-switch.php

```diff
@@ -0,0 +1,579 @@
+ *
+ * @param mixed $update_handler The WP_Ajax_Upgrader_Skin instance
+ * @param mixed $upgrader The Plugin_Upgrader instance
+ * @return string Extracted messages
+ */
+function extract_handler_messages( $update_handler, $upgrader ): string {
+	$handler_messages = '';
+	
+	if ( isset( $update_handler ) ) {
+		// Errors (WP_Ajax_Upgrader_Skin stores them)
+		if ( method_exists( $update_handler, 'get_errors' ) ) {
+			$errs = $update_handler->get_errors();
+			if ( $errs instanceof \WP_Error && $errs->has_errors() ) {
+				$handler_messages .= implode( "\n", $errs->get_error_messages() );
+			}
+		}
+		// Error messages string
+		if ( method_exists( $update_handler, 'get_error_messages' ) ) {
+			$em = $update_handler->get_error_messages();
+			if ( $em ) {
+				$handler_messages .= "\n" . $em;
+			}
+		}
+		// Upgrade messages (feedback/info)
+		if ( method_exists( $update_handler, 'get_upgrade_messages' ) ) {
+			$upgrade_msgs = $update_handler->get_upgrade_messages();
+			if ( is_array( $upgrade_msgs ) ) {
+				$handler_messages .= "\n" . implode( "\n", $upgrade_msgs );
+			} elseif ( $upgrade_msgs ) {
+				$handler_messages .= "\n" . (string) $upgrade_msgs;
+			}
+		}
+	}
+
+	// Fallback: if upgrader populated result with info, include it
+	if ( empty( $handler_messages ) && isset( $upgrader->result ) ) {
+		if ( is_wp_error( $upgrader->result ) ) {
+			$handler_messages = implode( "\n", $upgrader->result->get_error_messages() );
+		} else {
+			$handler_messages = is_scalar( $upgrader->result ) ? (string) $upgrader->result : print_r( $upgrader->result, true );
+		}
+	}
+
+	return trim( $handler_messages );
+}
+
+/**
+ * Log version switch attempt for debugging
+ *
+ * @param string $target_version Target version
+ * @param mixed  $result Installation result
+ * @param string $details Additional details
+ * @return void
+ */
+function log_version_switch_attempt( string $target_version, $result, string $details = '' ): void {
+	if ( function_exists( 'error_log' ) ) {
+		error_log( sprintf( 
+			'Code Snippets version switch failed. target=%s, result=%s, details=%s', 
+			$target_version, 
+			var_export( $result, true ), 
+			$details 
+		) );
+	}
+}
+
+/**
+ * Handle version switch request
+ *
+ * @param string $target_version Target version to switch to
+ * @return array Result array with success status and message
+ */
+function handle_version_switch( string $target_version ): array {
+	// Check user capabilities
+	if ( ! current_user_can( 'update_plugins' ) ) {
+		return create_error_response( __( 'You do not have permission to update plugins.', 'code-snippets' ) );
+	}
+
+	// Validate target version
+	$available_versions = get_available_versions();
+	$validation = validate_target_version( $target_version, $available_versions );
+	
+	if ( ! $validation['success'] ) {
+		return create_error_response( $validation['message'] );
+	}
+
+	// Check if already on target version
+	if ( get_current_version() === $target_version ) {
```

Comment:
```suggestion
	$is_already_on_target_version = get_current_version() === $target_version;
	if ( $is_already_on_target_version ) {
```

Please remove all the redundant comments from the file.

Reply:
done